code/game/mecha/mecha.dm

```diff
@@ -915,12 +937,16 @@
 	var/mob/living/brainmob = mmi_as_oc.brainmob
 	mmi_as_oc.mecha = src
 	occupant = brainmob
+	silicon_pilot = TRUE
 	brainmob.forceMove(src) //should allow relaymove
 	brainmob.reset_perspective(src)
 	brainmob.remote_control = src
 	brainmob.update_mobility()
 	brainmob.update_mouse_pointer()
-	icon_state = initial(icon_state)
+	if (silicon_icon_state)
+		icon_state = silicon_icon_state
+	else
+		icon_state = initial(icon_state)

```

Comment:
why is this check not just in update_icon?

Reply:
I added the check everywhere the original code initial'd the icon_state variable. I suppose I could just check the status of silicon_pilot and silicon_icon_state at the top of update_icon.

Reply:
mech code...

Reply:
Is it possible to check if an icon exists? If so, I could rename the empty icon to ripley-silicon, do an `if (silicon_pilot && <initial icon + "-silicon" exists>)`, and just remove the silicon_icon_state variable.

Reply:
no, besides the var is fine I'm just grossed out by the same check everywhere.

Reply:
It's slightly better now.